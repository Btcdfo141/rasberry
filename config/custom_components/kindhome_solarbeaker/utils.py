def log(msg):
	print(f"MATI: {msg}")

def log(fun, msg):
    print(f"MATI, {fun}: {msg}")
