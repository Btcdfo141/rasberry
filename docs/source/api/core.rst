.. _core_module:

:mod:`spencerassistant.core`
-------------------------

.. automodule:: spencerassistant.core
    :members: