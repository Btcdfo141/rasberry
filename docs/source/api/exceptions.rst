.. _exceptions_module:

:mod:`spencerassistant.exceptions`
-------------------------------

.. automodule:: spencerassistant.exceptions
    :members:
