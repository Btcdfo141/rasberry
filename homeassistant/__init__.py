"""Init file for Home Assistant."""


test2
