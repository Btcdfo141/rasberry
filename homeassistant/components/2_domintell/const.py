"""Constants for the Domintell Integration integration."""

DOMAIN = "2_domintell"
