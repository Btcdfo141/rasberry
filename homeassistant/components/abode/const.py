"""Constants for the Abode Security System component."""
import logging

LOGGER = logging.getLogger(__package__)

DOMAIN = "abode"
ATTRIBUTION = "Data provided by goabode.com"

DEFAULT_CACHEDB = "jaraco_abode_cache.pickle"
CONF_POLLING = "polling"
