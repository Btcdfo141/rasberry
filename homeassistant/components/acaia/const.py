"""Constants for component."""
from typing import Final

DOMAIN: Final = "acaia"
CONF_IS_NEW_STYLE_SCALE: Final = "is_new_style_scale"
