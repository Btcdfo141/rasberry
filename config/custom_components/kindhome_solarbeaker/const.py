from datetime import timedelta

from homeassistant.const import Platform

DOMAIN="kindhome_solarbeaker"
PLATFORMS = [Platform.COVER]

UPDATE_INTERVAL = timedelta(seconds=20)

SERVICE_UUID = "75c276c3-8f97-20bc-a143-b354244886d4"
TITLE="Kindhome Solarbeaker"

DATA_DEVICE = "device"
DATA_COOR = "coordinator"
