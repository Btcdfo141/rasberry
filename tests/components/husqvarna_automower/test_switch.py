"""Tests for switch platform."""
from datetime import timedelta
from unittest.mock import AsyncMock

from aioautomower.exceptions import ApiException
from aioautomower.model import MowerStates, RestrictedReasons
from aioautomower.utils import mower_list_to_dictionary_dataclass
from freezegun.api import FrozenDateTimeFactory
import pytest

from homeassistant.components.husqvarna_automower.const import DOMAIN
from homeassistant.core import HomeAssistant
from homeassistant.exceptions import HomeAssistantError

from . import setup_integration
from .const import TEST_MOWER_ID

from tests.common import (
    MockConfigEntry,
    async_fire_time_changed,
    load_json_value_fixture,
)


async def test_switch_states(
    hass: HomeAssistant,
    mock_automower_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
    freezer: FrozenDateTimeFactory,
) -> None:
    """Test switch state."""
    values = mower_list_to_dictionary_dataclass(
        load_json_value_fixture("mower.json", DOMAIN)
    )
    await setup_integration(hass, mock_config_entry)

    for state, restricted_reson, expected_state in [
        (MowerStates.RESTRICTED, RestrictedReasons.NOT_APPLICABLE, "on"),
        (MowerStates.IN_OPERATION, RestrictedReasons.NONE, "off"),
    ]:
        values[TEST_MOWER_ID].mower.state = state
        values[TEST_MOWER_ID].planner.restricted_reason = restricted_reson
        mock_automower_client.get_status.return_value = values
        freezer.tick(timedelta(minutes=5))
        async_fire_time_changed(hass)
        await hass.async_block_till_done()
        state = hass.states.get("switch.test_mower_1_park_until_further_notice")
        assert state.state == expected_state


@pytest.mark.parametrize(
    ("service", "aioautomower_command"),
    [
        ("turn_on", "park_until_further_notice"),
        ("turn_off", "resume_schedule"),
        ("toggle", "park_until_further_notice"),
    ],
)
async def test_lawn_mower_commands(
    hass: HomeAssistant,
    aioautomower_command: str,
    service: str,
    mock_automower_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
) -> None:
    """Test switch commands."""
    await setup_integration(hass, mock_config_entry)

    getattr(mock_automower_client, aioautomower_command).side_effect = ApiException(
        "Test error"
    )

    with pytest.raises(HomeAssistantError) as exc_info:
        await hass.services.async_call(
            domain="switch",
            service=service,
            service_data={"entity_id": "switch.test_mower_1_park_until_further_notice"},
            blocking=True,
        )
    assert (
        str(exc_info.value)
        == "Command couldn't be sent to the command queue: Test error"
    )
