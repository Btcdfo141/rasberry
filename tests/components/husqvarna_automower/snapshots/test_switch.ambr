# serializer version: 1
# name: test_switch[switch.test_mower_1_park_until_further_notice-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.test_mower_1_park_until_further_notice',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Park until further notice',
    'platform': 'husqvarna_automower',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'park_until_further_notice',
    'unique_id': 'c7233734-b219-4287-a173-08e3643f89f0_park_until_further_notice',
    'unit_of_measurement': None,
  })
# ---
# name: test_switch[switch.test_mower_1_park_until_further_notice-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Test Mower 1 Park until further notice',
    }),
    'context': <ANY>,
    'entity_id': 'switch.test_mower_1_park_until_further_notice',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
